src/app/api/village.ts

```diff
@@ -0,0 +1,11 @@
+import instance from './axiosInstance';
+
+export const Village = {
+    async getVillageUser(pageNo : number, take : number, sort? : string, animal? : string, nickname? : string) : Promise<any>{
+        const result = await instance.get(
+            `users?pageNo=${pageNo}&take=${take}&animal=${animal}&orderByField=${sort}&nickname=${nickname}`
+        )
+        console.log(result.data)
+        return result.data
+    }
```

Comment:
```suggestion
const type VillageType = {
   pageNo : number,
   take : number,
   ...
   nickname? : string,
}

export const Village = {
    async getVillageUser(params : VillageType) : Promise<any>{
        const result = await instance.get(`users`, {
                params: {
                   pageNo : params.pageNo,
                   take: params.take,
                   ...
                   nickname : params.nickname
                },
            } );
        console.log(result.data)
        return result.data
    }
```
`파라미터`값을 전달할때도 마찬가지로, 모든 정보를 `( )` 안쪽에 넣은 것 보다, `type`을 따로 분리해서 사용하시는게 관리하기도 편하고, 확인하기도 좋습니다. 따로 `types`폴더를 만들어서 관리하는 것 또한 하나의 방법입니다.

위에 하신 방법대로 해도 동작은 하겠지만, 가독성이 떨어질 수 있습니다. `axios`에서 지원하는 `params`를 활용해서 진행하시면, 편하게 진행이 가능합니다.

Reply:
추가적으로 `api`를 만드실 때, 주석을 사용해서 다른 개발자가 어떤 기능을 만들었는지 확인하게 하는 것 또한 협업 개발에서 중요합니다.